aws/credential_cache_test.go

```diff
@@ -211,6 +211,121 @@ func TestCredentialsCache_ExpireTime(t *testing.T) {
 	}
 }
 
+func TestCredentialsCache_AsyncRefresh(t *testing.T) {
+	orig := sdk.NowTime
+	defer func() { sdk.NowTime = orig }()
+	var mockTime time.Time
+	sdk.NowTime = func() time.Time { return mockTime }
+
+	cases := []struct {
+		Creds  func() Credentials
+		Called int32
+	}{
+		{
+			Called: 6,
```

Comment:
See my comment above, I wouldn't expect 6 calls here. It should only be when we are within the expiry window that an async refresh is kicked off.

Reply:
this test is indeed a bit confusing, the number of call here is to fetch token be it sync or async